uint32 centre_x
uint32 centre_y
